### PR TITLE
Replace pytz and tzlocal by zoneinfo

### PR DESCRIPTION
**Pull request by jspricke**
Requires Python 3.9

### CODE REVIEW

**Comment by thehunmonkgroup** [comment]
This will be welcome, but it's much too soon from a compatibility standpoint -- there are important distros not running 3.9 yet.

**Comment by jspricke** [comment]
Hi, https://github.com/GothenburgBitFactory/tasklib/pull/109 was merged and I added a backports dependency for old Python version. Do you think this is ready for merging?

File: vit/formatter/__init__.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import math
 from datetime import datetime
-from pytz import timezone
+from zoneinfo import ZoneInfo
```
Comment (thehunmonkgroup) [inline vit/formatter/__init__.py]: The documentation at https://pypi.org/project/backports.zoneinfo/ is suggesting this idiom for the fallback:

```python
try:
    import zoneinfo
except ImportError:
    from backports import zoneinfo
```

Is there some reason the official documentation suggestion is not being followed?

File: vit/formatter_base.py [hunk 2]
```diff
@@ -1,7 +1,6 @@
 from importlib import import_module
 from datetime import datetime, timedelta
-from tzlocal import get_localzone
-from pytz import timezone
+from zoneinfo import ZoneInfo
```
Comment (thehunmonkgroup) [inline vit/formatter_base.py]: Same as previous fallback question.

**Review by thehunmonkgroup — CHANGES_REQUESTED** [review]

**Comment by jspricke** [comment]
>The documentation at https://pypi.org/project/backports.zoneinfo/ is suggesting this idiom for the fallback:
>
>```python
>try:
>    import zoneinfo
>except ImportError:
>    from backports import zoneinfo
>```
>
>Is there some reason the official documentation suggestion is not being followed?

Fixed, thx.

**Comment by thehunmonkgroup** [comment]
Can you provide any clarity why ```pytz``` and ```tzlocal``` are still being installed despite not being listed in the requirements any more?

```
[:~/git/vit] jspricke-zoneinfo+ ± cat requirements.txt 
tasklib>=2.4.0
urwid>=2.1.0
backports.zoneinfo;python_version<"3.9"
[:~/git/vit] jspricke-zoneinfo+ ± pip install -r requirements.txt
Collecting tasklib>=2.4.0
  Using cached tasklib-2.4.3-py3-none-any.whl
Requirement already satisfied: urwid>=2.1.0 in /home/hunmonk/.pyenv/versions/3.7.4/lib/python3.7/site-packages (from -r requirements.txt (line 2)) (2.1.2)
Requirement already satisfied: backports.zoneinfo in /home/hunmonk/.pyenv/versions/3.7.4/lib/python3.7/site-packages (from -r requirements.txt (line 3)) (0.2.1)
Collecting pytz
  Using cached pytz-2021.3-py2.py3-none-any.whl (503 kB)
Collecting tzlocal
  Using cached tzlocal-4.1-py3-none-any.whl (19 kB)
Requirement already satisfied: pytz-deprecation-shim in /home/hunmonk/.pyenv/versions/3.7.4/lib/python3.7/site-packages (from tzlocal->tasklib>=2.4.0->-r requirements.txt (line 1)) (0.1.0.post0)
Requirement already satisfied: tzdata in /home/hunmonk/.pyenv/versions/3.7.4/lib/python3.7/site-packages (from pytz-deprecation-shim->tzlocal->tasklib>=2.4.0->-r requirements.txt (line 1)) (2021.5)
Installing collected packages: tzlocal, pytz, tasklib
Successfully installed pytz-2021.3 tasklib-2.4.3 tzlocal-4.1
```

**Comment by jspricke** [comment]
>Can you provide any clarity what ```pytz``` and ```tzlocal``` are still being installed despite not being listed in the requirements any more?
[..]
>Requirement already satisfied: pytz-deprecation-shim in /home/hunmonk/.pyenv/versions/3.7.4/lib/python3.7/site-packages (from tzlocal->tasklib>=2.4.0->-r requirements.txt (line 1)) (0.1.0.post0)
>Requirement already satisfied: tzdata in /home/hunmonk/.pyenv/versions/3.7.4/lib/python3.7/site-packages (from pytz-deprecation-shim->tzlocal->tasklib>=2.4.0->-r requirements.txt (line 1)) (2021.5)

Because the pip version of tasklib still depends on them.

**Comment by thehunmonkgroup** [comment]
Got it. So it seems sensible to wait until the next release of ```tasklib```, then we can properly bump the dependencies.

**Comment by jspricke** [comment]
* thehunmonkgroup ***@***.***> [2022-03-15 07:47]:
>Got it. So it seems sensible to wait until the next release of ```tasklib```, then we can properly bump the dependencies.

Just to ask, does it make any difference for vit? As long as the 
dependencies are not imported in vit they should not be used, right?

**Comment by thehunmonkgroup** [comment]
> Just to ask, does it make any difference for vit? As long as the
> dependencies are not imported in vit they should not be used, right?

Correct. I suppose we could commit this now and simply delay the 2.2.0 release until after we bump the ```tasklib``` deps. I think it just makes for a cleaner overall experience to coordinate that.

**Pull request merged**